dig.go

```diff
@@ -498,6 +524,31 @@ func (c *Container) Invoke(function interface{}, opts ...InvokeOption) error {
 	return nil
 }
 
+func (c *Container) verifyAcyclic() error {
+	// invert the c.providers map to satisfy verifyAcyclic's arguments
+	nodesToKeys := make(map[dot.CtorID][]key)
```

Comment:
If possible, I would like to avoid making `dot.CtorID` part of the business
logic like this. It exists to identify constructors for the GraphViz
visualization only.

If you change `detectCycles` to handle the `len(path) == 0` case (which it
doesn't with #208 because of the `path[0]`), you should be able to use that
directly here.

    visited := make(map[key]struct{})
    for _, n := range c.nodes {
        if err := detectCycles(n, c, nil /* path */, visited); err != nil {
            // ...
        }
    }

Reply:
This makes sense. I am going to wait for #208 to land to update this PR appropriately.